config/pitest-suppressions/pitest-regexp-suppressions.xml

```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<suppressedMutations>
+  <mutation unstable="false">
+    <sourceFile>RegexpCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.regexp.RegexpCheck</mutatedClass>
+    <mutatedMethod>processRegexpMatches</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.RemoveConditionalMutator_EQUAL_ELSE</mutator>
+    <description>removed conditional - replaced equality check with false</description>
+    <lineContent>if (!illegalPattern &amp;&amp; !checkForDuplicates) {</lineContent>
```

Comment:
Pitest survivals are not allowed at all. Please cover it by test.

Reply:
Relevant to my question above https://github.com/checkstyle/checkstyle/pull/15761#discussion_r1817314238. Pitest says it is not used, but you mentioned that some tests are failing. @mohitsatr could you recheck please?

Reply:
done.  ```if (!illegalPattern && !checkForDuplicates)``` is gone.